external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -294,32 +296,36 @@ impl<'env, 'map> Context<'env, 'map> {
     pub fn pop_warning_filter_scope(&mut self) {
         self.defn_context.pop_warning_filter_scope()
     }
+
+    pub fn check_feature(&self, package: Option<Symbol>, feature: FeatureGate, loc: Loc) -> bool {
+        self.env()
+            .check_feature(self.reporter(), package, feature, loc)
+    }
 }
 
 impl DefnContext<'_, '_> {
     pub(super) fn add_diag(&self, diag: Diagnostic) {
-        self.env.add_diag(&self.warning_filters_scope, diag);
+        self.reporter.add_diag(diag);
```

Comment:
I kept all of the various delegates I added in #20065, since it will make future refactors maybe a bit easier? 
I can remove them if there are strong opinions